HW2/P2/parallel_vector.pyx

```diff
@@ -80,12 +80,37 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    # data movement.  Be sure to avoid deadlock, and double-locking.
    ##########
    with nogil:
-       for r in range(repeat):
+       for r in prange(repeat, num_threads=4):
            for idx in range(src.shape[0]):
+               if src[idx] < dest[idx]: 
```

Comment:
Usually, you want to do a check again after acquiring the lock, since the condition might have changed between checking the condition and acquiring the lock